### PR TITLE
Add subscription styles switch

### PR DESCRIPTION
**Pull request by DaveAMA**
This PR adds the styles needed for the email subscription form found here on dropmark: http://daveama.dropmark.com/436337

The styles for the switch itself are a bit long, but I built them out so that the ✖️  and ✔️ icons can be easily swapped out in the markup for any other fa-icons should we want to in the future.  For the same reason I haven't added the icons using pseudo-elements, and instead I am including them as elements in the markup as follows: 

```
<label class="switch">
  <input type="checkbox">
  <div class="switch-slider">
    <div class="fa fa-check switch-slider__icon--left"></div>
    <div class="fa fa-times switch-slider__icon--right"></div>
  </div>
</label>
```

I've removed the `layout_components/subscriptions.scss` styles since I was able to replicate the styles using existing helper classes. (https://github.com/amaabca/my_ama/pull/1314)

@sineade during the review, can you confirm that I am using BEM correctly? I am still a bit shaky on some of our syntax there. 

🎨 

### PR Review Checklist
To be completed by the person who opens the PR. Use strikethroughs for items that are not applicable in list.
- [x] I have reviewed my own PR to check syntax & logic, removed unnecessary/old code, made sure everything aligns with our style guide and BEM.
- [ ] I've added new components to the style guide (or have a PR in to add them).
- [x] Any applicable version numbers have been updated.
- [x] I've tested on mobile, tablet, and desktop, as well as across all of the browsers we support.
- [ ] I've proof-read all text for legibility, proper grammar, punctuation, and capitalization (titlecase).
- [x] I have written a detailed PR message explaining what is being changed and why
- [x] I’ve linked to any relevant VSTS tickets
- [x] I’ve added the appropriate review symbols to my PR - (elephant) for dev review, (art) for design

### CODE REVIEW

File: app/assets/stylesheets/layout_components/forms.scss [hunk 1]
```diff
@@ -63,6 +63,93 @@ form {
...
+
+  &-slider {
+    position: absolute;
+    cursor: pointer;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    background-color: $brand-red;
+    -webkit-transition: 0.3s;
```
Comment (jessedoyle) [inline app/assets/stylesheets/layout_components/forms.scss]: This project uses https://github.com/postcss/autoprefixer, so we shouldn't need to manually add vendored attributes. Autoprefixer should automatically add this to our CSS for us!

We could get rid of all `-webkit` and `-ms` attributes in this PR. What do you think?

File: app/assets/stylesheets/layout_components/forms.scss [hunk 2]
```diff
@@ -63,6 +63,93 @@ form {
...
+    transition: 0.3s;
+    border-radius: 50%;
+  }
+
+  input:checked + .switch-slider {
+    background-color: $brand-green;
+
+    .switch-slider__icon{
+
```
Comment (jessedoyle) [inline app/assets/stylesheets/layout_components/forms.scss]: What do you think about dropping this newline?

File: app/assets/stylesheets/layout_components/forms.scss [hunk 3]
```diff
@@ -63,6 +63,93 @@ form {
...
+      &--right {
+        @extend .switch-slider__icon;
+        opacity: 0;
+        -webkit-transition-delay: 0s;
+        transition-delay: 0s;
+      }
+    }
+
+    &:before {
+    -webkit-transform: translateX(34px);
```
Comment (jessedoyle) [inline app/assets/stylesheets/layout_components/forms.scss]: This block of attributes appears to be missing 2 spaces of indentation.

**Comment by jessedoyle** [comment]
:book:

**Comment by jessedoyle** [comment]
:hankey:

File: app/assets/stylesheets/layout_components/forms.scss (hunk 1 above)
Comment (DaveAMA) [reply app/assets/stylesheets/layout_components/forms.scss]: Awesome, good to know

**Comment by DaveAMA** [comment]
👋 Thanks for the review @jessedoyle!

**Comment by jessedoyle** [comment]
:art::art::art:

**Pull request merged**